### PR TITLE
Studio: Allow import of wp-config.php

### PR DESCRIPTION
**Pull request by kozer**
Fixes https://github.com/Automattic/dotcom-forge/issues/8661

## Proposed Changes

When a user has a custom table_prefix, this is not currently respected, as we ignore `wp-config.php` file.
This PR solves that.

## Testing Instructions

1. Create a new site.
2. Create a backup from a site with a custom wp_prefix ( You can use the backup located U1WJZT58T/F07GH2T0UAE-slack-file )
3. Import the backup
4. Ensure that the correct tables get used after you start your local site.
5. Check your local site's `wp-config.php` file, and ensure that you see the correct `$table_prefix`.

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/lib/import-export/import/importers/importer.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ abstract class BaseBackupImporter extends BaseImporter {
 			await this.moveExistingDatabaseToTrash( dbPath );
 			await this.createEmptyDatabase( dbPath );
 			await this.importDatabase( rootPath, siteId, this.backup.sqlFiles );
+			await this.importWpConfig( rootPath );
```
Comment (sejas) [inline src/lib/import-export/import/importers/importer.ts]: Let's create the wp-config file before `importDatabase` to fix https://github.com/Automattic/studio/pull/437#issuecomment-2286057665

```suggestion
			await this.importWpConfig( rootPath );
			await this.importDatabase( rootPath, siteId, this.backup.sqlFiles );
```

File: src/lib/import-export/tests/import/validators/local-validator.test.ts [hunk 2]
```diff
@@ -65,7 +66,7 @@ describe( 'LocalValidator', () => {
 				'app/public/wp-admin/wp-admin.php',
 				'app/public/wp-admin/about.php',
 				'app/public/wp-includes/test.php',
-				'app/public/wp-content/wp-config.php',
```
Comment (sejas) [inline src/lib/import-export/tests/import/validators/local-validator.test.ts:68]: Nice!

**Review by sejas — APPROVED** [review]
I tested it and I confirm it works as expected.
Importing the wp-config file.

**Comment by jeroenpf** [comment]
@kozer I am concerned about the comment by @wojtekn ( p1723457289097709/1723450926.132199-slack-C04GESRBWKW ) 

If someone is using MySQL instead of SQLite, copying the wp-config file entirely will overwrite the DB credentials. 

If we want to prevent that we could read the imported wp-config and update the existing wp-config while excluding the db credentials. You could put the imported wp-config at a temporary location accessible by wp cli and then [list the options ](https://developer.wordpress.org/cli/commands/config/list/) and [update](https://developer.wordpress.org/cli/commands/config/set/) the existing wp-config from that list, skipping the DB options.

Or, we accept that importing on a MySQL Studio site will break the connection. What you think?

**Comment by kozer** [comment]
@jeroenpf , I meanly merged it to resolve https://github.com/Automattic/dotcom-forge/issues/8625#issuecomment-2286095643
I don't know what is the best way to go here. Using MySQL, is not part of Studio currently, so we will fix something for a case we don't support as a tool.
However I see the point.
@wojtekn , what do you think? Do you think it's better to work on a solution like the one Jeroen proposed?
Also, I would love @sejas  opinion as well.
If we decide to go with another solution, I'll create an issue and work on it tomorrow, first thing.

**Comment by sejas** [comment]
It's a great question. Not only for the the credentials but for other custom code that could live in wp-config.php

I think we should not allow importing sites into MySQL sites or in the other hand we need to create another importer that will execute the correct wp-cli commands and all the other differences.
- We have open this issue: https://github.com/Automattic/dotcom-forge/issues/8606

It's also worth mentioning that users accept a modal to replace all the files and database.

Screenshot 2024-08-13 at 17 14 37

**Comment by fluiddot** [comment]
> I think we should not allow importing sites into MySQL sites or in the other hand we need to create another importer that will execute the correct wp-cli commands and all the other differences.

Yep, I agree. When I was exploring https://github.com/Automattic/dotcom-forge/issues/8606, I also thought we should disable import/export for sites using MySQL (p1723104840771259/1723043486.040399-slack-C04GESRBWKW) as it seems the best way to go until we give support. I can open a PR with the fix.

**Pull request merged**